### ISSUE TITLE
npm install fails in the latest version of LEL

### ISSUE THREAD

**Issue by ananyaarun** [publiclab/leaflet-environmental-layers#180]
## Description
Due to redundant code in 
- index.html
- LeafletEnvironmantalLayers.js
npm install fails to build resulting in errors on console.

**Comment by ananyaarun** [comment]
@sagarpreet-chadha @gauravano , while trying to solve this, i redid my local installation and now i am getting this error ... not sure if it due to npm version problem.

![error](https://user-images.githubusercontent.com/32260628/58043656-5fbd6980-7b5b-11e9-9d9c-42d63af7b382.png)

I installed both popper and leaflet as shown in the warnings using npm install.
I even installed the optional dependency fsevents.
after running npm audit fix there are still 3 errors that say it needs to be manually fixed,

![error1](https://user-images.githubusercontent.com/32260628/58043787-bcb91f80-7b5b-11e9-9fbf-28c152712515.png)

I never had such issues before when i installed LEL, any idea where i am missing something ?

**Comment by sagarpreet-chadha** [comment]
Hey , i will take a look at this later today .
Was nodes modules created after you ran npm install ? I think it is related to Linux OS .

**Comment by ananyaarun** [comment]
Yes it was, and I deleted that folder every time I ran npm install again while trying.

**Comment by ananyaarun** [comment]
I tried to clear cache ` npm cache clean --force` and tried again as well , but still doesn't seem to work.
Could the package-lock.json be causing some issues here ? I deleted this as well and tried again but still there is some errors :confused:

**Comment by sagarpreet-chadha** [comment]
Hey @ananyaarun , this is weird error .
 There were some vulnerablities issue that i just resolved . Can you try again to run npm install  ? Make sure you have latest code on your PC . 
Do tell me if the error still persists ? Thanks !

**Comment by ananyaarun** [comment]
Hey @sagarpreet-chadha 
My branch is even with master and i cloned the repo again and ran npm install ... still not working though and i get the same old errors on running npm audit fix.

![notworking](https://user-images.githubusercontent.com/32260628/58123301-281cf300-7c29-11e9-8ed2-b7e0bc360213.png)

Whats interesting is I tried to build someone elses fork , which is a few commits behind master branch ... and it gives the same output on running npm install , but does work !! I can see the LEL layers on this .

![working](https://user-images.githubusercontent.com/32260628/58123414-5f8b9f80-7c29-11e9-9ee5-2e62dc6d04df.png)

**Comment by sagarpreet-chadha** [comment]
Okay , Can you send screenshot of the console in browser ?

**Comment by ananyaarun** [comment]
Here is the screen shot , Oh ok I'll try running npm install again on the latest version.
![console](https://user-images.githubusercontent.com/32260628/58151287-d3a86080-7c86-11e9-905c-932f8e1a813f.png)

**Comment by sagarpreet-chadha** [comment]
Thanks @ananyaarun , I am now pretty sure that this PR https://github.com/publiclab/leaflet-environmental-layers/pull/130 was the issue . 
You can send the PR for this :) 
PS : Make sure that leaflet.js is added as a script in head , and all scripts are loaded in the head , and of course removing the redundant code will solve this .

**Comment by ananyaarun** [comment]
@sagarpreet-chadha , I did this but looks like there is a code redundancy issue in dist/LeafletEnvironmentalLayers.js lines 21298 to 21896 ... var polygon is defined twice due to which some variables and functions are not getting defines correctly. Can u confirm if this is the case actually ?
I'm trying to fix this too now :)

**Comment by sagarpreet-chadha** [comment]
Hey,
I think leaflet.js script file is missing in index.html , can you try to put that just below head tag (at the top of other script files) .
Also we do not change the code in dist folder , if there is corresponding code in src folder , you can open an issue and tag me there . After changing src files we use command grunt build to make file in dist folder .

From the error in console , hopefully i think it is missing script file error . Let's see , thanks :)

**Comment by ananyaarun** [comment]
do you mean these as given on the leaflet website ?
 <link rel="stylesheet" href="https://unpkg.com/leaflet@1.5.1/dist/leaflet.css"
    integrity="sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ=="
    crossorigin=""/>
    <script src="https://unpkg.com/leaflet@1.5.1/dist/leaflet.js"
    integrity="sha512-GffPMF3RvMeYyc1LWMHtK8EbPv0iNZ8/oTtHPx9/cc2ILxQ+u905qIwdpULaqDkyBKgOaB57QTMg7ztg8Jm2Og=="
    crossorigin=""></script>

Sorry wasn't aware of how the dist folder is structured ! will keep that in mind. Thnaks !

**Comment by rarrunategu1** [comment]
Hello @sagarpreet-chadha.  I'm having the same issues where npm install is not installing leaflet or popper.  Should I just follow the instruction you gave to ananyaarun?  
![image](https://user-images.githubusercontent.com/29240939/58446987-d1a03080-80d0-11e9-9838-c5842dee1dc3.png)

**Comment by sagarpreet-chadha** [comment]
Hi @rarrunategu1 , i have just merged the PR which solves this issue . You can pull the latest code and try again . If the issue still persists , feel free to ping me . Thanks .

**Comment by rarrunategu1** [comment]
Hello,  I had the same issue even after the update but I solved it by running npm install -g npm instead and now I have a running application with a map.  
These were the errors I was getting. 
![image](https://user-images.githubusercontent.com/29240939/58487274-4d33c900-8135-11e9-9494-22a1885d234e.png)
I noticed that the leaflet it's trying to install is 1.5.1 but the package.json is referencing 1.3.1.  Could that be part of the problem?  I'm not sure if it is.  I also looked up the error on popper.js and from what I'm researching, the error pops up because bootstrap 4.0 already comes with popper.  
![image](https://user-images.githubusercontent.com/29240939/58488201-f3340300-8136-11e9-9be5-10bec24406f6.png)
